pkg/kubelet/dockershim/docker_container.go

```diff
@@ -162,11 +162,20 @@ func (ds *dockerService) CreateContainer(_ context.Context, r *runtimeapi.Create
 
 	hc.SecurityOpt = append(hc.SecurityOpt, securityOpts...)
 
+	cleanupInfo, err := ds.applyPlatformSpecificDockerConfig(r, &createConfig)
+	if err != nil {
+		return nil, err
+	}
+
 	createResp, err := ds.client.CreateContainer(createConfig)
 	if err != nil {
 		createResp, err = recoverFromCreationConflictIfNeeded(ds.client, createConfig, err)
```

Comment:
If we end up in this block, note that the container name is regenerated at https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/kubelet/dockershim/helpers.go#L299 . Will it be safer if `applyPlatformSpecificDockerConfig` => `applyGMSAConfig` directly uses `createConfig.Name` and cleans up/regenerates in case of a conflict?

Reply:
At this point might as well make the registry value purely random, doing that.